File: src/main/java/com/ptah/common/CacheService.java
package com.ptah.common;

public interface CacheService {
    <T> T getByKey(String key, Class<T> type);

    String getByKey(String key);

    void setValue(String key, Object value);

    void setValue(String key, Object value, long timeout);

    boolean hasKey(String key);
}

File: src/main/kotlin/com/ptah/common/CacheService.kt
package com.ptah.common

interface CacheService {
    fun <T> getByKey(key: String?, type: Class<T>): T
    fun getByKey(key: String?): String
    fun setValue(key: String, value: Any)
    operator fun setValue(key: String, value: Any, timeout: Long)
    fun hasKey(key: String): Boolean
}
